libs/blocks/video/video.js

```diff
@@ -11,4 +14,16 @@ export default function init(a) {
   const videoElem = document.body.querySelector(`source[src=".${pathname}"]`)?.parentElement;
   applyHoverPlay(videoElem);
   a.remove();
+};
+
+export default async function init(a) {
+  if (a.textContent.includes('no-lazy')) {
+    loadVideo(a);
+  } else {
+    createIntersectionObserver({
```

Comment:
It looks to me that the intersection observer is not always needed; I think it makes more sense to load it dynamically similar to focal point in marquee - https://github.com/adobecom/milo/blob/main/libs/blocks/marquee/marquee.js#L53

Reply:
This ticket is for performance. I think Video will be still loaded background if it is not handled in video.js.
Do you see any downside to have intersection observer on video?

Reply:
I don't see a downside to having an intersection observer for video, all I'm saying is that `createIntersectionObserver` is only used when video is lazy loaded.
```js
  if (a.textContent.includes('no-lazy')) {
    loadVideo(a);
  } else {
    const { createIntersectionObserver } = await import('../../utils/utils.js'');
    createIntersectionObserver({
      el: a,
      options: { rootMargin: `${ROOT_MARGIN}px` },
      callback: loadVideo,  
    });
  }
```

Reply:
I think having video lazy loaded (w/ intersection observer) as default is a good way to improve our performance in overall.
And author can add `no-lazy` in case author doesn't want with the current logic so, it is good to go as it is.

Reply:
I agree that video should be lazy loaded and the intersection observer is good for performance but what I mentioned and highlighted with the code was that maybe we should import the `createIntersectionObserver` method only when authors want to load the video lazy.

Reply:
I am not sure if I am understanding your comment 100% but, I believe it should be almost no impact on page performance having another import `utils.js` since it should be loaded before it got here (at scripts.js). And what I am trying to say is authors should want to load the video lazy as default, and in case they want no-lazy, it won't hit import createIntersectionObserver anyway since it is only under `else` condition.

Reply:
Hey Sean, no worries, this was mainly a suggestion to showcase best practices. We've had performance problems in the past and since there weren't any existing good examples in the code, the PRs were without awareness.